lib/backends/rq/rqprocessor.js

```diff
@@ -214,32 +233,65 @@ function _processItemMethod (tree, parentPath, name, remotePrefix, localPrefix,
           completeRequest();
           return;
         }
+       var chunkOffset = 0;
+	   var fileSize ;
+	   var chunkSizeFixed = 1024*1024*chunksInMb;
+	   uploadNextChunk = function () {
+		 if(chunkOffset >= fileSize)
+		 {
+		 /* File uploaded completely. Complete the request now*/
+			completeRequest();
+			return;
+		 }
+		 var req = getRequest();
+		 var form = req.form();
+		 if(method == 'PUT')
+		 {
+		   logger.debug('Updating the file %s, Use post for CreateAssetServelet',serverPath);
+		   req.method = 'POST';
+		   form.append('replaceAsset','true');
+		 }
+		 var chunkSize = chunkOffset+ chunkSizeFixed < fileSize ? chunkSizeFixed : fileSize - chunkOffset;
+		 logger.debug('Uploading chunks to the file % of size %s Chunk curent offset %s chunksize %s',serverPath, fileSize, chunkOffset, chunkSize);
+		 read = fs.createReadStream(localPath,{'start':chunkOffset, 'end':chunkSize+chunkOffset-1});
+		 read.on('error', function (err) {
+		 	handleError(localPath, method, err);
+		  }); 
+		/* if filesize is less than the chunk, upload file at once*/
+		 if(chunkSizeFixed < fileSize) {
+			form.append('file@Offset', chunkOffset);
+		 }
+		 form.append('chunk@Length', chunkSize);
+		 form.append('file@Length', fileSize);
+  	     if(chunkOffset >= fileSize) {
+			form.append('file@Completed', 'true');
+		 }
+		 form.append('file', read);
+		 webutils.monitorTransferProgress(read, serverPath, localPath, fileSize, function (progress) {
+		    /* Add the chunk offset, thats what is already uploaded*/
+			progress.read = progress.read + chunkOffset - chunkSize;
```

Comment:
It's hard to tell just by looking at the code, but does the file's upload progress still show correctly in AEM Desktop's asset status dialog?

Reply:
Yes. I saw the logs and its showing correctly. Since we are uploading in chunks, we need to add already uploaded chunks to the progress, other wise the % will always be a percent of chunk